libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/SpecTranslate/Core.hs

```diff
@@ -103,3 +96,27 @@ withCtx ctx m = do
   case runSpecTransM ctx m of
     Right x -> pure x
     Left e -> throwError e
+
+-- | OpaqueErrorString behaves like unit in comparisons, but contains an
+-- error string that can be displayed.
+newtype OpaqueErrorString = OpaqueErrorString (NonEmpty Text)
+  deriving (Generic, Show, Semigroup)
+
+instance Eq OpaqueErrorString where
+  _ == _ = True
+
+instance ToExpr OpaqueErrorString where
+  toExpr (OpaqueErrorString x) = App (T.unpack . T.unlines $ toList x) []
+
+instance NFData OpaqueErrorString
+
+showOpaqueErrorString :: ToExpr a => a -> OpaqueErrorString
+showOpaqueErrorString = OpaqueErrorString . (NE.:| []) . T.pack . showExpr
+
+unComputationResult :: Agda.ComputationResult Text a -> Either OpaqueErrorString a
+unComputationResult (Agda.Success x) = Right x
+unComputationResult (Agda.Failure e) = Left (OpaqueErrorString $ e NE.:| [])
+
+unComputationResult_ :: Agda.ComputationResult Void a -> Either e a
+unComputationResult_ (Agda.Success x) = Right x
```

Comment:
Would it make more sense to not return `Either` here? It would require usage of `Right` in the place where it is used, but it would feel more natural then having a function that always returns `Right`
```suggestion
unComputationResult_ :: Agda.ComputationResult Void a -> a
unComputationResult_ (Agda.Success x) = x
```

Reply:
I guess it would work, but I also like how I can just use it in place of `unComputationResult` in the rules that don't have any predicate failures. I think it'll be removed anyways once I refactor the conformance framework, so I'd rather just leave it the way it is right now.